discord/automod.py

```diff
@@ -0,0 +1,459 @@
+
+    def to_dict(self) -> AutoModerationRulePayload:
+        ret: AutoModerationRulePayload = {
+            'id': str(self.id),
+            'guild_id': str(self.guild.id),
+            'name': self.name,
+            'creator_id': str(self.creator_id),
+            'event_type': self.event_type.value,
+            'trigger_type': self.trigger.type.value,
+            'trigger_metadata': self.trigger.to_metadata_dict(),
+            'actions': [action.to_dict() for action in self.actions],
+            'enabled': self.enabled,
+            'exempt_roles': [str(role_id) for role_id in self.exempt_role_ids],
+            'exempt_channels': [str(channel_id) for channel_id in self.exempt_channel_ids],
+        }  # type: ignore # trigger types break the flow here.
+
+        return ret
+
+    @property
+    def creator(self) -> Optional[Member]:
+        """Optional[:class:`Member`]: The member that created this rule."""
+        return self.guild.get_member(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: The roles that are exempt from this rule."""
+        result = []
+        get_role = self.guild.get_role
+        for role_id in self.exempt_role_ids:
+            role = get_role(role_id)
+            if role is not None:
+                result.append(role)
+
+        return utils._unique(result)
+
+    @property
+    def exempt_channels(self) -> List[Union[GuildChannel, Thread]]:
+        """List[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The channels that are exempt from this rule."""
+        it = filter(None, map(self.guild._resolve_channel, self.exempt_channel_ids))
+        return utils._unique(it)
+
+    @property
+    def actions(self) -> List[AutoModRuleAction]:
+        """List[:class:`AutoModRuleAction`]: The actions that are taken when this rule is triggered."""
+        return [AutoModRuleAction.from_data(action) for action in self._actions]
+
+    def is_exempt(self, object: Union[GuildChannel, Thread, Role], /) -> bool:
+        """This method will check if an object is exempt from the automod rule.
+
+        Parameters
+        -----------
+        object: Union[:class:`abc.GuildChannel`, :class:`Thread`, :class:`Role`]
+            The object which we are checking.
+
+        Returns
+        --------
+        :class:`bool`
+        """
+        return object.id in self.exempt_channel_ids or object.id in self.exempt_role_ids
+
+    async def edit(
+        self,
+        *,
+        name: str = MISSING,
+        event_type: AutoModRuleEventType = MISSING,
+        actions: List[AutoModRuleAction] = MISSING,
+        enabled: bool = MISSING,
+        exempt_roles: List[Role] = MISSING,
+        exempt_channels: Sequence[Snowflake] = MISSING,
+        reason: str = MISSING,
+    ) -> Self:
+        """|coro|
+
+        Edit an existing auto moderation rule. All parameters of this method are optional.
```

Comment:
"Edit*s* this auto moderation rule" (see other edit methods), no need to mention params being optional